doc: add-on v1.0.1 update (#10)
nrfconnect/ncs-zigbee-r22: docs/release-notes.rst

+-------------------+------------------+-----------------------+---------------------+
| |addon| version   | |NCS| version    | ZBOSS stack version   | NCP host version    |
+===================+==================+=======================+=====================+
| 1.0.1             | 2.9.0            | 3.11.8.0              | 2.2.7               | 
+-------------------+------------------+-----------------------+---------------------+
| 1.0.0             | 2.9.0            | 3.11.7.0              | 2.2.6               | 
+-------------------+------------------+-----------------------+---------------------+

.. _zigbee_release:

|addon| v1.0.1 - 15/07/2025
***************************

> This is a production ready release of |addon|.

Reviewer: ```suggestion
This is a production-ready release of |addon|.
```

> * Added:
>  
>   * Fix security vulnerability.
>   * Fix building when FACTORY_RESET is disabled.
>   * Improvement for handling transmission when acknowledgment is not received.
>   * Adjustment of NVRAM partition size for nRF52833.

Reviewer: +should it mention what kind of vulnerability was fixed?

```suggestion
* Updated:

  * Fixed security vulnerability.
  * Fixed build issues when FACTORY_RESET is disabled.
  * Improved transmission handling in scenarios where acknowledgments are not received.
  * Adjusted the NVRAM partition size to optimize performance on the nRF52833 platform.
```

> 

Reviewer: ```suggestion

|addon| v1.0.0 - 07/03/2025
***************************

This is a production-ready release of |addon|.
 
* Updated:
 
  * Fixed the logging of negative temperatures in the :ref:`Thingy:53: Zigbee weather station <zigbee_weather_station_app_include>` application.
  * Fixed an issue where an assert occurs on an empty record during NVRAM migration after upgrading to ZBOSS from NCS 2.7.0 or higher.
  * Fixed a fatal error after changing the ``ZB_DEV_REJOIN_TIMEOUT_MS`` setting.
  * Fixed the process for setting the endpoint field in the ZCL device callback.
```
